lib/_element.js

```diff
@@ -344,6 +352,70 @@ function element () {
     }))
   }
 
+  Element.prototype.compareDocumentPosition = function (other) {
+    function getFirstNodeByOrder (nodes, nodeOne, nodeTwo) {
+      return nodes.reduce(function (result, node) {
+        return result !== false
+          ? result
+          : node === nodeOne
+            ? nodeOne
+            : node === nodeTwo
+              ? nodeTwo
+              : node.childNodes
+                ? getFirstNodeByOrder(node.childNodes, nodeOne, nodeTwo)
+                : false
+      }, false)
+    }
+
+    function isSourceContainTarget (source, target) {
```

Comment:
How about `isAncestor`? As in `source isAncestor of target, I feel like that reads a little better as a name?